pkg/orchestrator/pipelines.go

```diff
@@ -80,5 +81,37 @@ func (s *PipelineOrchestrator) Delete(ctx context.Context, id string) error {
 	if len(pl.ProcessorIDs) != 0 {
 		return ErrPipelineHasProcessorsAttached
 	}
-	return s.pipelines.Delete(ctx, pl.ID)
+	return po.pipelines.Delete(ctx, pl.ID)
+}
+
+func (po *PipelineOrchestrator) UpdateDLQ(ctx context.Context, id string, dlq pipeline.DLQ) (*pipeline.Instance, error) {
+	pl, err := po.pipelines.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	if pl.ProvisionedBy != pipeline.ProvisionTypeAPI {
+		return nil, cerrors.Errorf("pipeline %q cannot be updated: %w", pl.ID, ErrImmutableProvisionedByConfig)
+	}
+	// TODO lock pipeline
+	if pl.Status == pipeline.StatusRunning {
+		return nil, pipeline.ErrPipelineRunning
+	}
+
+	// cast orchestrator to a ConnectorOrchestrator to get access to the plugin
+	// config validate function
+	err = (*ConnectorOrchestrator)(po).Validate(
```

Comment:
Why does the cast work btw?

Reply:
All orchestrators have the same underlying type [`base`](https://github.com/ConduitIO/conduit/blob/54025b01872bb2dccc5e133fd6ae8f0d7e719552/pkg/orchestrator/orchestrator.go#L62), they just define different methods on top. See how the orchestrators are created [here](https://github.com/ConduitIO/conduit/blob/54025b01872bb2dccc5e133fd6ae8f0d7e719552/pkg/orchestrator/orchestrator.go#L55-L58).

Here's a simplified example of what's happening: https://go.dev/play/p/aQQRxZaMnHC